test/gtest/ucp/ucp_test.h

```diff
@@ -266,4 +266,58 @@ std::ostream& operator<<(std::ostream& os, const ucp_test_param& test_param);
     UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, self,   "self") \
     UCP_INSTANTIATE_TEST_CASE_TLS(_test_case, tcp,    "tcp")
 
+
+/**
+ * Auxiliary macro to instantiate the parameterized test case for all
+ * transport combinations with GPU memory awareness
+ *
+ * @param _test_case     Test case class, derived from ucp_test.
+ * @param _gpu_name      GPU memory type name (cuda/rocm).
+ * @param _gpu_copy_tls  GPU memory copy TLs (cuda_copy/rocm_copy/etc).
+ * @param _gpu_ipc_tls   GPU IPC TLs (cuda_ipc/rocm_ipc/etc).
+ */
+#define _UCP_INSTANTIATE_TEST_CASE_GPU_AWARE(_test_case, _gpu_name, \
```

Comment:
maybe add alias in UCP transports: "gpu" for all gpu transports?
any reason to instantiate only specific GPU transports? we should use different GPU transports by enumerating on memory type in the test itself

Reply:
> maybe add alias in UCP transports: "gpu" for all gpu transports?

done, but this alias can be used only for `shm_ib_ipc` to enable CUDA/ROCm IPC TLs in order to not use CUDA/ROCm IPC instead of TCP/IB/uGNI/etc TLs that we want to test

> any reason to instantiate only specific GPU transports? we should use different GPU transports by enumerating on memory type in the test itself

yes, this is a good idea, done